mlrun/runtimes/kubejob.py

```diff
@@ -235,10 +235,25 @@ def deploy(
             # get the clone target dir in case it was enriched due to loading source
             self.spec.clone_target_dir = get_in(data, "data.spec.clone_target_dir")
             ready = data.get("ready", False)
+            target_image = (
+                data.get("data", {}).get("spec", {}).get("build", {}).get("image")
+            )
             if not ready:
-                logger.info(
-                    f"Started building image: {data.get('data', {}).get('spec', {}).get('build', {}).get('image')}"
+                logger.info(f"Started building image: {target_image}")
+
+            if (
+                target_image.endswith(":latest")
+                and self.spec.image_pull_policy
+                and self.spec.image_pull_policy != "Always"
+            ):
+                # When image pull policy is not set and image tag is latest, k8s defaults to Always - no need to warn
+                logger.warning(
+                    "Target image tag is 'latest' but image_pull_policy is set and not 'Always', "
+                    "this may cause the function to use a stale image",
```

Comment:
the warning is about the pod not pulling the latest image because there is already another `latest` image available locally (due to previous run or whatever). therefore "stale image" refers to the "latest" available locally and not the updated latest in the docker registry. of course this is true for all tags, but we warn for latest because it's the default tag and it's easy to miss.